### PR TITLE
move hidden into module

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by burrbull** [comment]
cc @rust-embedded/tools

**Comment by thejpster** [comment]
v0.29: https://docs.rs/rp2040-pac/latest/rp2040_pac/

image

This patch: 

image

**Comment by thejpster** [comment]
I like these changes. But I wonder if there's also scope to try and add some information to this page:

image

File: src/generate/register.rs [hunk 1]
```diff
@@ -138,19 +139,28 @@ pub fn render_register_mod(
...
     if can_read {
         let desc = format!("Register `{}` reader", register.name);
         mod_items.extend(quote! {
             #[doc = #desc]
             pub type R = crate::R<#regspec_ident>;
         });
-        methods.push("read");
+        write!(
+            doc,
+            "\nThis register you can {} with [`R`].",
```
Comment (thejpster) [inline src/generate/register.rs:161]:
```suggestion
            "You can {} this register and get [`R`]. ",
```

File: src/generate/register.rs [hunk 2]
```diff
@@ -159,17 +169,29 @@ pub fn render_register_mod(
...
         });
+        let mut methods = Vec::new();
         methods.push("write_with_zero");
         if can_reset {
             methods.push("reset");
             methods.push("write");
         }
+        write!(
+            doc,
+            "\nThis register you can {} with [`W`].",
```
Comment (thejpster) [inline src/generate/register.rs:180]:
```suggestion
            "You can {} this register using [`W`]. ",
```

File: src/generate/register.rs [hunk 3]
```diff
@@ -159,17 +169,29 @@ pub fn render_register_mod(
...
+                .iter()
+                .map(|m| method(m))
+                .collect::<Vec<_>>()
+                .join(", ")
+        )?;
     }
 
     if can_read && can_write {
-        methods.push("modify");
+        write!(doc, "\nAnd {} with [`R`] and [`W`].", method("modify"))?;
```
Comment (thejpster) [inline src/generate/register.rs:190]:
```suggestion
        write!(doc, "You can also {} this register using [`R`] and [`W`]. ", method("modify"))?;
```

File: src/generate/register.rs [hunk 4]
```diff
@@ -159,17 +169,29 @@ pub fn render_register_mod(
...
     }
 
+    doc.push_str("\nSee [API](https://docs.rs/svd2rust/#read--modify--write-api).");
```
Comment (thejpster) [inline src/generate/register.rs:193]:
```suggestion
    doc.push_str("\n\nSee [API](https://docs.rs/svd2rust/#read--modify--write-api) for more details.");

File: src/generate/register.rs [hunk 5]
```diff
@@ -138,19 +139,28 @@ pub fn render_register_mod(
     );
 
     let mut mod_items = TokenStream::new();
-    let mut methods = vec![];
 
     let can_read = access.can_read();
     let can_write = access.can_write();
     let can_reset = properties.reset_value.is_some();
 
+    let mut doc = format!("{description}\n");
```
Comment (thejpster) [inline src/generate/register.rs:147]:
```suggestion
    let mut doc = format!("{description}\n\n");
```

**Review by thejpster — CHANGES_REQUESTED** [review]
Slightly change the wording.

**Comment by burrbull** [comment]
@thejpster 
Now it looks like
изображение
and
изображение
accordingly.

**Comment by thejpster** [comment]
Can we remove the '\n' between the lines? If they have '\n' between them I think they should be a bulleted list. Otherwise I think it should be one paragraph with no new-lines in it.

i.e.

```quote
You can read this register and get somemod::R. You can reset, write, write_with_zero this register and get somemod::W. You can also modify this register.
```

rather than

```quote
You can read this register and get somemod::R.
You can reset, write, write_with_zero this register and get somemod::W. 
You can also modify this register.
```

I also propose expanding on *See API* a little bit.

But these are minor things really. I really like these changes!

**Comment by burrbull** [comment]
done

**Comment by thejpster** [comment]
LGTM. Ship it.

**Review by thejpster — APPROVED** [review]

**Review by thejpster — APPROVED** [review]

**Comment by thejpster** [comment]
So good I managed to approve it twice :/

**Pull request merged**